standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSMetricsListener.java

```diff
@@ -101,7 +101,14 @@ public void onAllocWriteId(AllocWriteIdEvent allocWriteIdEvent, Connection dbCon
       Table table = getTable(allocWriteIdEvent);
 
       if (MetaStoreUtils.isNoAutoCompactSet(table.getParameters())) {
-        Metrics.getOrCreateGauge(MetricsConstants.WRITES_TO_DISABLED_COMPACTION_TABLE).incrementAndGet();
+        int noAutoCompactSet =
+            Metrics.getOrCreateGauge(MetricsConstants.WRITES_TO_DISABLED_COMPACTION_TABLE).incrementAndGet();
+        if (noAutoCompactSet >=
+            MetastoreConf.getIntVar(getConf(),
+                MetastoreConf.ConfVars.COMPACTOR_NUMBER_OF_DISABLED_COMPACTION_TABLES_THRESHOLD)) {
+          LOGGER.warn("Number of tables where the compaction is turned off is: " + noAutoCompactSet);
```

Comment:
This might be clearer: "There has been a write to a table where auto-compaction is disabled (tblproperties ("no_auto_compact"="true"))...
And definitely log the db and table name, so users can find it and re-enable auto-compaction.